bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -390,10 +407,62 @@ private BluetoothGattDescriptor getTinybDescriptorByUUID(String uuid) {
         return null;
     }
 
+    private synchronized void removeForInactivity() {
+        if (device != null && !device.getConnected()) {
+            logger.debug("Removing device '{}' due to inactivity", device.getAddress());
+            try {
+                disableNotifications();
+                device.remove();
+            } catch (BluetoothException ex) {
+                if (ex.getMessage().contains("Does Not Exist")) {
+                    // this happens when the underlying device has already been removed
+                    // but we don't have a way to check if that is the case beforehand so
+                    // we will just eat the error here.
+                } else {
+                    logger.debug("Exception occurred when trying to remove inactive device '{}': {}", address,
+                            ex.getMessage());
+                }
+            }
+        }
+    }
+
     /**
      * Clean up and release memory.
      */
     public void dispose() {
+        inactiveCleanupJob.cancel();
```

Comment:
also set the inactiveCleanupJob field to null here.

Reply:
done